x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsDest.java

```diff
@@ -20,40 +21,50 @@
 public class DataFrameAnalyticsDest implements Writeable, ToXContentObject {
 
     public static final ParseField INDEX = new ParseField("index");
+    public static final ParseField RESULTS_FIELD = new ParseField("results_field");
+
+    private static final String DEFAULT_RESULTS_FIELD = "ml";
 
     public static ConstructingObjectParser<DataFrameAnalyticsDest, Void> createParser(boolean ignoreUnknownFields) {
         ConstructingObjectParser<DataFrameAnalyticsDest, Void> parser = new ConstructingObjectParser<>("data_frame_analytics_dest",
-            ignoreUnknownFields, a -> new DataFrameAnalyticsDest((String) a[0]));
+            ignoreUnknownFields, a -> new DataFrameAnalyticsDest((String) a[0], (String) a[1]));
         parser.declareString(ConstructingObjectParser.constructorArg(), INDEX);
+        parser.declareStringOrNull(ConstructingObjectParser.optionalConstructorArg(), RESULTS_FIELD);
```

Comment:
`declareStringOrNull` I think implies that the passed argument could be explicitly `null`. This means `results_field: null` is a valid payload. This does not make sense to me. It seems to be that it being optional is good so we can set the default, but allowing it to be explicitly null and then overwriting it seems strange to me.

Reply:
Ah, yes! Good spot! I got tricked to think it was the method to use for optional fields.